scaleway/resource_redis_cluster_test.go

```diff
@@ -298,6 +302,249 @@ func TestAccScalewayRedisCluster_Settings(t *testing.T) {
+					//resource.TestCheckResourceAttr("scaleway_redis_cluster.main", "private_network.0.service_ips.1", "192.168.1.0/20"),
+					resource.TestCheckResourceAttrSet("scaleway_redis_cluster.main", "private_network.0.id"),
+					resource.TestCheckResourceAttrSet("scaleway_redis_cluster.main", "private_network.0.endpoint_id"),
+					resource.TestCheckTypeSetElemAttrPair("scaleway_redis_cluster.main", "private_network.0.id", "scaleway_vpc_private_network.pn", "id"),
+					resource.TestCheckNoResourceAttr("scaleway_redis_cluster.main", "private_network.1.service_ips.0"),
+					resource.TestCheckNoResourceAttr("scaleway_redis_cluster.main", "private_network.1.id"),
+					resource.TestCheckNoResourceAttr("scaleway_redis_cluster.main", "private_network.1.endpoint_id"),
+				),
+			},
+			{
+				// And finally we remove the private network to check that we still have a public network
+				Config: `
+					resource "scaleway_vpc_private_network" "pn" {
+						name = "private-network"
+					}
+					resource "scaleway_vpc_private_network" "pn2" {
+						name = "private-network-2"
+					}
+					resource "scaleway_redis_cluster" "main" {
+						name = "test_redis_endpoints"
+						version = "6.2.6"
+						node_type = "MDB-BETA-M"
+						user_name = "my_initial_user"
+						password = "thiZ_is_v&ry_s3cret"
+						cluster_size = 1
+						depends_on = [
+							scaleway_vpc_private_network.pn,
+							scaleway_vpc_private_network.pn2,
+						]
+					}
+				`,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayRedisExists(tt, "scaleway_redis_cluster.main"),
+					resource.TestCheckResourceAttr("scaleway_redis_cluster.main", "name", "test_redis_endpoints"),
+					resource.TestCheckResourceAttr("scaleway_redis_cluster.main", "version", "6.2.6"),
+					resource.TestCheckResourceAttr("scaleway_redis_cluster.main", "node_type", "MDB-BETA-M"),
+					resource.TestCheckResourceAttr("scaleway_redis_cluster.main", "user_name", "my_initial_user"),
+					resource.TestCheckResourceAttr("scaleway_redis_cluster.main", "password", "thiZ_is_v&ry_s3cret"),
+					resource.TestCheckResourceAttr("scaleway_redis_cluster.main", "cluster_size", "1"),
+					resource.TestCheckResourceAttrSet("scaleway_redis_cluster.main", "public_network.0.id"),
+					resource.TestCheckResourceAttrSet("scaleway_redis_cluster.main", "public_network.0.ips.#"),
+					resource.TestCheckResourceAttrSet("scaleway_redis_cluster.main", "public_network.0.port"),
+					resource.TestCheckNoResourceAttr("scaleway_redis_cluster.main", "private_network.0.id"),
+				),
+			},
+			{
+				// Extra step just to be sure that the cluster is deleted before the Private Networks
+				Config: `
+					resource "scaleway_vpc_private_network" "pn" {
+						name = "private-network"
+					}
+					resource "scaleway_vpc_private_network" "pn2" {
+						name = "private-network-2"
+					}
+				`,
+			},
+			//{
+			//	// PHASE TWO: CLUSTER WITH 3 NODES
```

Comment:
Could you split your test in two test: One with EndpointStandalone and a second test with cluster.

Also you don't need 3 nodes, just 2 is enough to enable cluster if I understand correctly.